dissect/target/filesystem.py

```diff
@@ -1030,44 +1038,58 @@ def map_dir(self, vfspath, realpath):
             vfsroot = fsutil.join(vfspath, relroot)
             directory = self.makedirs(vfsroot)
 
-            for d in dirs:
-                self.makedirs(fsutil.join(vfsroot, d))
+            for dir_ in dirs:
+                vfs_dir = fsutil.join(vfsroot, dir_)
+                self.makedirs(vfs_dir)
 
-            for f in files:
-                fullpath = os.path.join(root, f)
-                directory.add(f, MappedFile(self, fsutil.join(vfsroot, f), fullpath))
+            for file_ in files:
+                vfs_file_path = fsutil.join(vfsroot, file_)
+                real_file_path = os.path.join(root, file_)
+                directory.add(file_, MappedFile(self, vfs_file_path, real_file_path))
 
-    def map_file(self, vfspath, realpath):
+    def map_file(self, vfspath: str, realpath: str):
         """Map a file from the host machine into the VFS."""
         vfspath = fsutil.normalize(vfspath)
-        self.map_file_entry(vfspath, MappedFile(self, vfspath, realpath))
+        if vfspath[-1] == "/":
+            raise AttributeError(f"Can not map a file onto a directory: {vfspath}")
```

Comment:
```suggestion
            raise AttributeError(f"Can't map a file onto a directory: {vfspath}")
```
Maybe it could be a feature that you can, but it takes the filename from the real path? Obviously that won't be possible for `map_file_fh`.

Reply:
Sounds good, but can you make a ticket for it then?

Reply:
Done